src/util/tr.js

```diff
@@ -499,6 +496,29 @@ const _tr = (msg, {
       const body = "<div class='wzrkPPdscr' style='color:" + textColor + "'>" + descriptionText + '<div></td></tr></table></div>'
       html = css + title + body
     }
+    if (displayObj.preview && displayObj['custom-editor']) {
+      iframe.sandbox = 'allow-scripts allow-popups allow-popups-to-escape-sandbox'
+      /* eslint-disable no-cond-assign */
+      var properties = {}
+      var styleRegex = /<style[^>]*>([^<]*)<\/style>/g
+      var ctClassRegex = /\.CT_(?:Box|Banner)\s*{([^}]*)}/g // Regex to match either .CT_Box or .CT_Banner
+      var cssRegex = /([\w-]+)\s*:\s*([^;}\s]+)(?:;|$)/g // Regex to extract property-value pairs
+      var styleContent, cssBlock, cssMatch
+
+      // Extract style blocks
+      while (styleContent = styleRegex.exec(html)) {
```

Comment:
did not understand this code, why are we assigning things in a while loop?

Reply:
This part is used to extract all the '<style>' block from html.
The while loop runs as long as styleRegex.exec(html) finds a match, processing each style block one at a time.